include/OpenSpaceToolkit/Astrodynamics/Flight/System/SatelliteDynamics.hpp

```diff
@@ -0,0 +1,161 @@
+        } ;
+
+        typedef std::vector<double> StateVector ; // Container used to hold the state vector
+        typedef std::function<void(const StateVector&, StateVector&, const double)> DynamicalEquationFuncCallback ; // Function pointer type for returning dynamical equation's pointers
+
+                                SatelliteDynamics                           (   const   Environment&                anEnvironment,
+                                                                                const   SatelliteSystem&            aSatelliteSystem,
+                                                                                const   State&                      aState,
+                                                                                const   SatelliteDynamics::StateVectorType&  aStateVectorType                   = SatelliteDynamics::StateVectorType::PositionVelocity   ) ;
+
+                                SatelliteDynamics                           (   const   SatelliteDynamics&          aSatelliteDynamics                          ) ;
+
+        SatelliteDynamics*      clone                                       ( ) const ;
+
+        bool                    operator ==                                 (   const   SatelliteDynamics&          aSatelliteDynamics                          ) const ;
+
+        bool                    operator !=                                 (   const   SatelliteDynamics&          aSatelliteDynamics                          ) const ;
+
+        friend std::ostream&    operator <<                                 (           std::ostream&               anOutputStream,
+                                                                                const   SatelliteDynamics&          aSatelliteDynamics                          ) ;
+
+        bool                    isDefined                                   ( ) const ;
+
+        void                    print                                       (           std::ostream&               anOutputStream,
+                                                                                        bool                        displayDecorator                            =   true ) const ;
+
+        SatelliteDynamics::StateVectorType getStateVectorType                               ( ) const ;
+
+        State                   getState                                    ( ) const ;
+
+        void                    setState                                    (   const   State&                      aState                                      ) ;
+
+        SatelliteDynamics::DynamicalEquationFuncCallback  accessDynamicalEquations ( ) const ;
+
+        static String           StringFromStateVectorType                   (   const   SatelliteDynamics::StateVectorType& aStateVectorType                            ) ;
+
+    private:
+
+        mutable Environment     environment_ ;
+        Shared<const Frame>     gcrfSPtr_ ;
```

Comment:
Is it necessary to keep a pointer to the frame in private of that class? 🤔

Reply:
I guess I didn't want to be instantiating a new gcrf pointer at every dynamics equation loop iteration for memory reasons. What do you think @lucas-bremond @Derollez

Reply:
Fair to me!